src/ExplainProvenanceSLD.h

```diff
@@ -249,6 +253,73 @@ class ExplainProvenanceSLD : public ExplainProvenance {
         }
     }
 
+    std::string measureRelation(std::string relName) override {
+        auto rel = prog.getRelation(relName);
+
+        if (rel == nullptr) {
+            return "No relation found\n";
+        }
+
+        auto size = rel->size();
+        int skip = size / 10;
+
+        if (skip == 0) skip = 1;
+
+        std::stringstream ss;
+
+        auto before_time = std::chrono::high_resolution_clock::now();
+
+        int numTuples = 0;
+        int proc = 0;
+        for (auto& tuple : *rel) {
+            auto tupleStart = std::chrono::high_resolution_clock::now();
+
+            if (numTuples % skip != 0) {
+                numTuples++;
+                continue;
+            }
+
+            std::vector<RamDomain> currentTuple;
+            for (size_t i = 0; i < rel->getArity() - 2; i++) {
+                RamDomain n;
+                if (*rel->getAttrType(i) == 's') {
+                    std::string s;
+                    tuple >> s;
+                    n = prog.getSymbolTable().lookupExisting(s.c_str());
+                } else {
+                    tuple >> n;
+                }
+
+                currentTuple.push_back(n);
+            }
+
+            RamDomain ruleNum;
+            tuple >> ruleNum;
+
+            RamDomain levelNum;
+            tuple >> levelNum;
+
+            std::cout << "Tuples expanded: "
+                      << explain(relName, currentTuple, ruleNum, levelNum, 20)->getSize();
+            numTuples++;
+            proc++;
+
+            auto tupleEnd = std::chrono::high_resolution_clock::now();
+            auto tupleDuration =
+                    std::chrono::duration_cast<std::chrono::duration<double>>(tupleEnd - tupleStart);
+
+            std::cout << ", Time: " << tupleDuration.count() << std::endl;
```

Comment:
I'm not sure what you're measuring here, but if you don't need to flush the before then outputting a `\n` instead of std::endl will avoid delays due to the flushing. You flush the buffers outside the for loop already, so it all eventually gets printed.